src/qibolab/instruments/rfsoc/driver.py

```diff
@@ -71,6 +74,18 @@ def connect(self):
     def disconnect(self):
         """Empty method to comply with Instrument interface."""
 
+    def prepare_interpolated_channels(self, sequence: list[dict]):
+        """Subtract and set mixer frequency for interpolated channel."""
+        for pulse in sequence:
+            mix_freq = self.ports(pulse["dac"]).mixer_frequency
```

Comment:
Is the channel not enough to identify the Port? (currently, of course)

Reply:
What do you mean? I am using just the "pulse.dac" do identify the port (?)

Reply:
`dac` is definitely a non-standard attribute, even in 0.1.x

https://github.com/qiboteam/qibolab/blob/80d13169c66447fef1bdfe5b8511dc36b549d09d/src/qibolab/pulses.py#L747-L786

Reply:
I know they are defined in Qibosoq, but the closer the Qibosoq is to Qibolab the better (that's why I was wondering about the `.channel` attribute)

Reply:
I see, I wanted to refer to this part of the platform:
```python
channels |= Channel("L3-22_qd", port=controller.ports(0))  # drive
```
I thought that it could become something like:
```python
channels |= Channel("L3-22_qd", port=controller.ports(0))  # drive

controller.ports(0).mixer_frequency = 100  # Hz
```

With channels I could do something like:

```python
pulse.channel.port.mixer_frequency = 100  # Hz
```

I'm fine with it, eventually

Reply:
Sorry to reply so late: exactly, I meant something like the latter snippet.

I know that, when you're preparing the platform, you need to know the details of your physical object, so you have to reason in terms of ADCs and DACs.
It's just that this could be confined to the platform definition.

Then, whichever physical structure you have, you will reason in terms of channels and ports.
Where *channels* are more or less abstract (even more in 0.2), but *ports* really encode the physical details of your object, though you can retrieve them even without specific platform knowledge.